test/watcher_test.js

```diff
@@ -139,6 +162,65 @@ describe('Watcher', function() {
       expect(builderBuild).to.have.been.called;
     });
 
+    it('on change, rebuild is invoked and cancel is invoked from the build', function() {
+      const pluginsCalled = {};
+
+      const waitForPlugin1 = defer();
+      const waitForPlugin2 = defer();
```

Comment:
I don't understand why we need these deferred's at all. At the moment, they are both resolved at the same time. Can you explain what purpose they serve?